web/client/components/background/PreviewIcon.jsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+
+require('./css/previewicon.css');
+
+const PreviewIcon = React.createClass({
+    propTypes: {
+        side: React.PropTypes.number,
+        frame: React.PropTypes.number,
+        margin: React.PropTypes.number,
+        src: React.PropTypes.string,
+        vertical: React.PropTypes.bool,
+        layer: React.PropTypes.object,
+        currentLayer: React.PropTypes.object,
+        onToggle: React.PropTypes.func,
+        onClose: React.PropTypes.func,
+        setLayer: React.PropTypes.func
+    },
+    getDefaultProps() {
+        return {
+            side: 50,
+            frame: 4,
+            margin: 5,
+            src: '',
+            vertical: false,
+            layer: {},
+            currentLayer: {},
+            onToggle: () => {},
+            onClose: () => {},
+            setLayer: () => {}
+        };
+    },
+    render() {
+        const containerClass = this.props.vertical ? 'background-preview-icon-container-vertical' : 'background-preview-icon-container-horizontal';
+        let type = this.props.layer.visibility ? containerClass + ' bg-primary' : containerClass + ' bg-body';
```

Comment:
Use const, let is not needed. You can do everything without calculating type twice